mewarpx/examples/capacitive_discharge_2d.py

```diff
@@ -0,0 +1,203 @@
+            N_INERT=self.N_INERT, exclude_collisions=['charge_exchange']
+        )
+
+        #######################################################################
+        # Neutral plasma injection                                            #
+        #######################################################################
+
+        self.vol_emitter = emission.UniformDistributionVolumeEmitter(
+            T=self.T_ELEC
+        )
+
+        self.plasma_injector = emission.PlasmaInjector(
+            emitter=self.vol_emitter, species1=self.electrons,
+            species2=self.ions, npart=2 * self.SEED_NPPC * self.NX * self.NZ,
+            T_2=self.T_INERT, plasma_density=self.PLASMA_DENSITY
+        )
+
+        #######################################################################
+        # Initialize run and print diagnostic info                            #
+        #######################################################################
+
+        mwxrun.init_run()
+
+        #######################################################################
+        # Add diagnostics                                                     #
+        #######################################################################
+
+        self.text_diag = diag_base.TextDiag(
+            diag_steps=self.DIAG_STEPS, preset_string='perfdebug'
+        )
+
+        self.field_diag = FieldDiagnostic(
+            diag_steps=self.DIAG_STEPS, barrier_slices=[0],
+            save_pdf=False, style='roelof', min_dim=2.0
+        )
+
+        # array to save ion density
+        self.rho_array = np.zeros(self.NZ + 1)
+
+    def _get_rho_ions(self):
+        rho_data = mwxrun.get_gathered_rho_grid('he_ions', False)
+        if mwxrun.me == 0:
+            self.rho_array += (
+                np.mean(rho_data[:,:,0], axis=0) / constants.q_e
+                / self.DIAG_STEPS
+            )
+
+    def run_sim(self):
+
+        if self.steps is not None:
+            mwxrun.simulation.step(self.steps)
+            self.text_diag.print_performance_summary()
+        else:
+            mwxrun.simulation.step(self.MAX_STEPS - self.DIAG_STEPS)
+            callbacks.installafterstep(self._get_rho_ions)
+            mwxrun.simulation.step(self.DIAG_STEPS)
+
+            if mwxrun.me == 0:
+                np.save('avg_rho_data.npy', self.rho_array)
+
+##########################
+# parse input parameters
+##########################
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    '-s', '--steps', help='set the number of simulation steps manually',
+    type=int
+)
+parser.add_argument(
+    '-n', help='Test number to run (1 to 4)', required=False, type=int,
+    default=1
+)
+args, left = parser.parse_known_args()
+sys.argv = sys.argv[:1]+left
+
+if args.n < 1 or args.n > 4:
+    raise AttributeError('Test number must be an integer from 1 to 4.')
+if args.steps:
+    CASE_NUM = 0
+    steps = args.steps
+else:
+    CASE_NUM = args.n - 1
+    steps = None
```

Comment:
Will it never make sense to run a fixed number of steps of some case other than 1, for performance testing, memory footprint measurement, etc?

Reply:
Yeah maybe. Good point. I'll take away the hard coding of case when `steps` are specified.